### ISSUE TITLE
Asynchronous provider APIs call

### ISSUE THREAD

**Issue by patata3000** [CityofSantaMonica/mds-provider-services#61]
Are you planning to make the API calls asynchronous? Would you accept a PR that'd do so?

**Comment by thekaveman** [comment]
Absolutely. The real API work happens in our [`mds-provider`](/CityofSantaMonica/mds-provider) library, which this repository wraps up as executable Docker containers. There was some [early](/CityofSantaMonica/mds-provider#13) [discussion](/CityofSantaMonica/mds-provider#46) along those lines in that repo.

If you'd like to take a look at implementing that, I would start from the [mds-0.3](/CityofSantaMonica/mds-provider/tree/mds-0.3) branch in `mds-provider`. This will be merged imminently, and contains a significant amount of cleanup in all aspects of the library, particularly the API client such that supporting async calls should be relatively straight-forward.
